src/mpi/request/waitall.c

```diff
@@ -59,7 +58,7 @@ int MPIR_Waitall_impl(int count, MPIR_Request * request_ptrs[], MPI_Status array
             }
         }
         MPID_Progress_end(&progress_state);
-    } else if (requests_property & MPIR_REQUESTS_PROPERTY__NO_GREQUESTS) {
```

Comment:
This change concerns me a little. Why aren't we checking for this optimization any more?

Reply:
We now handle all extended generalized request outside of MPID_Waitall, there is also a change in the caller to make this unecessary.

Reply:
We now handle all extended generalized request outside of MPID_Waitall, there is also a change in the caller to make this unecessary.